client/src/Components/Layouts/SingleCaptains/Popups/Popup.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from "react";
+import { Modal } from "antd";
+
+class Popup extends Component {
+  state = {
+    visible: this.props.visible,
+    name: {
+      error: '',
+      value: ''
+    },
+    phone: {
+      error: '',
+      value: '',
+    }
+  };
+
+  handleOk = e => {
+    this.props.visibleFun("singleCaptain", "editVisibilty")(e);
+  };
+
+  handleCancel = e => {
+    this.props.visibleFun("singleCaptain", "editVisibilty")(e);
+  };
+  componentWillReceiveProps(props) {
+    this.setState({ visible: props.visible });
+  }
+
+  handleChange = e => {
+    const value = e.target.value;
+    const name = e.target.name;
+    if (name === 'name') {
+      if (/^([a-z]|\s)+$/.test(value)) {
+        this.setState({ [name]: { error: '', value } })
+      } else {
+        this.setState({ [name]: { value, error: 'الرجاء ملىء الحقل بحروف' } })
+      }
+    } else {
+      if (/^\+[0-9]+$/.test(value)) {
+        this.setState({ [name]: { error: '', value } })
+      } else {
+        this.setState({ [name]: { value, error: 'الرجاء ملىء الحقل بارقام' } })
+      }
+    }
+  };
+
+  render() {
+    const { name, phone } = this.state;
+    return (
+      <Modal
+        title="تعديل الطلب"
+        visible={this.state.visible}
```

Comment:
@naremanhilles I think it would be a simpler design if you used `this.props.visible` directly, instead of copying props.visible into state.visible on mount and willReceiveProps.